src/library/dlgtagfetcher.cpp

```diff
@@ -250,45 +293,63 @@ void DlgTagFetcher::fetchTagFinished(
 }
 
 void DlgTagFetcher::slotNetworkResult(
-        int httpError, const QString& app, const QString& message, int code) {
+        int httpError,
+        const QString& app,
+        const QString& message,
+        int code) {
     m_networkResult = httpError == 0 ? NetworkResult::UnknownError : NetworkResult::HttpError;
     m_data.m_pending = false;
-    QString strError = tr("HTTP Status: %1");
-    QString strCode = tr("Code: %1");
-    httpStatus->setText(strError.arg(httpError) + "\n" + strCode.arg(code) + "\n" + message);
-    QString unknownError = tr("Mixxx can't connect to %1 for an unknown reason.");
-    cantConnectMessage->setText(unknownError.arg(app));
-    QString cantConnect = tr("Mixxx can't connect to %1.");
-    cantConnectHttp->setText(cantConnect.arg(app));
+
+    if (m_networkResult == NetworkResult::UnknownError) {
+        QString unknownError = tr("Can't connect to %1 for an unknown reason.");
+        loadingProgressBar->setFormat(unknownError.arg(app));
+    } else {
+        QString cantConnect = tr("Can't connect to %1.");
+        loadingProgressBar->setFormat(cantConnect.arg(app));
+    }
+
+    btnRetry->setEnabled(true);
     updateStack();
 }
 
 void DlgTagFetcher::updateStack() {
+    m_progressBarValue = kPercentOfPopulation;
```

Comment:
Is it correct to always reset the progress bar value here? If yes, this should be reflected in the function name.

Reply:
This is only available in the first population. We update the stack in case of "Import Metadata From Musicbrainz" pressed or track changed via buttons.

The progress bar value changes after each WebTask, but it never updates stack. 

Stack is updated just in case of failure (the bar set to maximum in this case).
When the fetching is success (it is also set to maximum).
When it is started (down to the percent of population).

As far as I tried, that was the correct way to populate the progress bar with the percent of population.

Reply:
Ok, than we have a naming issue here. updateStack() sounds as it is regularly called, but for my understanding it is called only once and kPercentOfPopulation is the initial of the progress bar progress.

Reply:
I couldn't find a better name for kPercentOfPopulation but kInitialPercentOfProgress sounds better IMHO. 

For the updateStack(), I can't be sure about the naming, it can be reloadStack() maybe or refreshStack(). It actually called many time during a few track fetching process, If we consider changing tracks, having results or not found errors. 

On the other hand if we think about a single Tag fetching progress, it is called once when the progress is finished, so we can call it also showResults(). So think in that way, It can be no result -on fetching and network error- or results -finished successfully-.

Reply:
No clear answer to this question can be an indicator, that some refactoring is required...

Reply:
It looks like as if the progress Is reset to the initial value each time updateStack() is called. 
How about this: 
* Rename the function to a more generic updateView() Or such.
* rewrite the function that it is obvious that it can be called repeated at any time.
* fispose  m_progressBarValue and replace it by an instant calculation of progress taken from the the other state and progress variables we already have.
* encapsulate the acces to the progress bar by one or two functions, and removevalm other accesses to it.

Reply:
How about I separate this function into two? Such as createView() and updateView().

This looks like this function is actually doing two things, creating the stack and meanwhile updates the stacks without touching the created stack which is difficult to maintain.

Reply:
Since there is only one QTreeWidget and it is filled by the results from web it needs to be cleared in each update stack called. But the problem is the actual track metadata is also considered as results. So clearing the results after each updateStack() can clear the actual track metadata, which we don't want to.

So when we start a new fetching,  this happens:
Fetching Starts.   -> createView()
Track Changes    -> createView()
User Click Retry  -> createView()
This can be the first state of the tag fetcher, the related buttons are disabled. Track tag is listed and it never deletes. Progress bar can be set to 0 in this state.

Network Error.        -> updateView()
Fetching Success  -> updateView()
These doesn't clear the results, it will just add them. We can update the progress bar in this state.

Meanwhile, If this works, I am not sure but this will be more easy to list the results after each success musicbrainz result, but I guess this can be postponed for the benefit of cover art fetcher.